src/Writer/ErrorLog.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Zend\Log\Writer;
+
+use Zend\Log\Formatter\Simple as SimpleFormatter;
+
+class ErrorLog extends AbstractWriter
+{
+    /**
+     * @var int
+     */
+    protected $mode;
+
+    /**
+     * @var string|null
+     */
+    protected $destination;
+
+
+    /**
+     * ErrorLog constructor.
+     * @param array|null|\Traversable $options
+     */
+    public function __construct($options = [])
+    {
+        parent::__construct($options);
+
+        $this->mode = isset($options['mode']) ? $options['mode']: 0 ;
```

Comment:
Cast `$options` to an array before attempting to pull values from it. Do so as follows:

```php
if ($options instanceof Traversable) {
    $options = iterator_to_array($options);
}

if (! is_array($options) {
    $options = [];
}
```

(You'll need to import `Traversable` at the top of the file.)

Additionally, add a space before the `:` in your ternary statements, please.